strongbox-web-core/src/main/java/org/carlspring/strongbox/security/authentication/suppliers/NpmLoginAuthenticationSupplier.java

```diff
@@ -0,0 +1,80 @@
+package org.carlspring.strongbox.security.authentication.suppliers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.carlspring.strongbox.authentication.api.password.PasswordAuthentication;
+import org.carlspring.strongbox.controllers.layout.npm.NpmArtifactController;
+import org.carlspring.strongbox.controllers.layout.npm.NpmUser;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.MediaType;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@Component
+@Order(4)
+public class NpmLoginAuthenticationSupplier implements AuthenticationSupplier
+{
+
+    @Inject
+    private ObjectMapper objectMapper;
+
+    @CheckForNull
+    @Override
+    public Authentication supply(@Nonnull HttpServletRequest request)
+    {
+        NpmUser npmUser = deserializeNpmUser(request);
+
+        if(isValidNpmUser(npmUser))
+        {
+            return new PasswordAuthentication(npmUser.getName(), npmUser.getPassword());
+        }
+        else
+        {
+            throw new BadCredentialsException("invalid.credentials");
+        }
+
+    }
+
+    @Override
+    public boolean supports(@Nonnull HttpServletRequest request)
+    {
+        return "PUT".equalsIgnoreCase(request.getMethod()) &&
+                request.getContentType()!=null &&
+                request.getContentType().contains(MediaType.APPLICATION_JSON_VALUE) &&
+                request.getRequestURI().contains("/-/user/org.couchdb.user:");
```

Comment:
Cosmetic: Could you please following the project's coding convention:
```suggestion
        return "PUT".equalsIgnoreCase(request.getMethod()) &&
                                      request.getContentType()!=null &&
                                      request.getContentType().contains(MediaType.APPLICATION_JSON_VALUE) &&
                                      request.getRequestURI().contains("/-/user/org.couchdb.user:");
```